addons/account/models/account_invoice.py

```diff
@@ -839,7 +839,7 @@ def message_new(self, msg_dict, custom_values=None):
         if journal:
             values['journal_id'] = journal.id
         # Passing `type` in context so that _default_journal(...) can correctly set journal for new vendor bill
-        invoice = super(AccountInvoice, self.with_context(type=values.get('type'))).message_new(msg_dict, values)
+        invoice = super(AccountInvoice, self.with_context(type=values.get('type'), journal_type="purchase")).message_new(msg_dict, values)
```

Comment:
```suggestion
        invoice = super(AccountInvoice, self.with_context(type=values.get('type'), journal_type=journal.type)).message_new(msg_dict, values)
```
with 
```python
        journal = self.env['account.journal'].search([
            ('type', 'in', ('purchase', 'sale')), ('alias_name', 'in', alias_names)
        ], limit=1)

```
This must have been something forgotten in https://github.com/odoo/odoo/pull/31655 because it makes sense to receive mails on journals of type `sale` since that task.

Reply:
@william-andre Done :+1:

Reply:
You need to change the `search` on `account.journal` too :wink:

Reply:
re-Done :wink: